File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> v(26,0);
        int left =0;
        int maxFreq=0;
        int maxLen=0;
        
        for(int right=0;right<s.size();right++){
            v[s[right] - 'A']++;

            maxFreq = max(maxFreq, v[s[right]-'A']);

            if((right-left+1) - maxFreq>k){
                v[s[left] - 'A']--;
                left++;
            }

            maxLen = max(maxLen, right-left+1);
        }
        return maxLen;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    int mDepth=0;

    public void mxDepth(TreeNode root, int depth){

        
        if(root==null){
            mDepth= Math.max(depth, mDepth);
            return;
        }

        depth++;

        
        mxDepth(root.left, depth);
        mxDepth(root.right, depth);
        
    }
    public int maxDepth(TreeNode root) {
        
        mxDepth(root, 0);

        return mDepth;
        
    }
}
